pkg/operator/provisioner.go

```diff
@@ -201,3 +207,19 @@ func processMonAddresses(monAddresses []string) []string {
 	}
 	return monAddrs
 }
+
+func createImageName(pvName string) string {
+	// generate a UUID for our image name
+	u := string(uuid.NewUUID())
+
+	// image name should be under 100 chars to support kernels older than 4.7
+	// when the RBD kernel module converts the image name to an OID, it will use "rbd_id.<imageName>",
+	// so we have to leave room for that "rbd_id." prefix too.
+	pvNameMaxLen := imageNameMaxLen - len(rbdIDPrefix) - len(imageNamePrefix) - len(u) - 2 // 2 hyphens
+	if len(pvName) > pvNameMaxLen {
+		// the PV name is too long, truncate it before including it in the final image name
+		pvName = pvName[:pvNameMaxLen]
+	}
+
+	return fmt.Sprintf("%s-%s-%s", imageNamePrefix, pvName, u)
```

Comment:
I'm not sure I have great answers to some of those questions since this was @kokhang's implementation first, although I think he was following some patterns used by the RBD in tree plug in.  However, I would guess that
1. we don't *need* a prefix, it's just a nice identifier to see at first glance which images were created by a dynamic provisioner.
1. Rook could indeed be put in the name, perhaps instead of k8s.
1. The cluster name isn't really necessary since images are only per cluster so including the cluster name would be somewhat redundant.
1. We don't have any control of the PV name coming in.  Depending on k8s environment, it could be different and I don't know what the rules are or if there's anything guaranteeing uniqueness.  Therefore, I'd be hesitant to not add a UUID of our own in, because I don't have any confidence that the PV name is unique itself.